hca/util/__init__.py

```diff
@@ -236,13 +237,12 @@ def store_refs(d):
     @property
     def swagger_spec(self):
         if not self._swagger_spec:
```

Comment:
I wonder if it would be beneficial to guard this section with a lock here so multiple threads racing through this at the same time don't all download the remote spec? It's a common pattern in my code to download the files in a bundle in parallel. All those threads start at the typically same time and go through the client initialization at the same time.

Reply:
That might save some time, but how much is debatable as the swagger definition is relatively small (<100kb) and the simultaneous fetch would only occur if the definition was not already cached + the time incurred synchronizing access to this property. Either way, I would say that it's out of the scope of this ticket and requires further investigation.

Reply:
> out of scope

Not really. This ticket is about thread safety, so locks fall into that category. I don't know if you have used locks before but I'm talking a about a three line change here. One line to import threading locks, one line to initialize the lock instance and one line for the `with lock` context manager invocation.

> That might save some time

It won't save time as the downloads happen concurrently. It saves server round-trips and lambda invocations. Considering that it is only three lines of code, I think it's worth it.